### ISSUE TITLE
Changes to FormFields that mutate arrays and objects don't trigger validation

### ISSUE THREAD

**Issue by jordanholliday** [lchemy/ng-forms#1]
Noticed this while working with our `CheckboxGroupComponent`.  The equality check below, excerpted from the `FormField` directive, won't catch changes to arrays and objects, so the validation doesn't run 😞

For now I'm binding `form.validate()` to the group's `valueChange` as a quick work around. 

```javascript
	protected prevValue: T | undefined;
	ngDoCheck(): void {
		let currentValue: T | undefined = this.getModelValue();
		if (currentValue === this.prevValue) {
			return;
		}

		this.prevValue = currentValue;
		this.valueChange.emit(currentValue);
		this.container.validate();
	}
```

**Comment by jessezhang91** [comment]
Good catch, I'll update it to do a shallow equality check if it's an object or an array.

**Comment by jessezhang91** [comment]
Addressed in 71b645740442b291a54775f8f1df6fac144677be
